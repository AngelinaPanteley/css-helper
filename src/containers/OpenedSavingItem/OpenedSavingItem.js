import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../../store/actions/index';
import styles from './OpenedSavingItem.scss';
import { calcStyles } from '../../shared/utility';
import Preview from '../../components/Preview/Preview';
import Pattern from '../../components/Code/Pattern/Pattern';

class OpenedSavingItem extends Component {
  constructor(props) {
    super(props);

    this.settings = this.props.settings[this.props.item.editorName];
    this.previewClass = this.settings.className;

    this.state = {
      styleValues: calcStyles(
        this.settings.style,
        props.item.controlValues,
        this.settings.controls
      ),
    }
  }

  openCopyHint = () => {
    this.props.openHint('Copied.');
  }

  render() {
    const wrapperStyles = [styles.Wrapper];
    if (this.props.isOpen) {
      wrapperStyles.push(styles.Open);
    } else {
      wrapperStyles.push(styles.Close);
    }

    let styleString = '';
    for (let styleKey in this.state.styleValues) {
      styleString = `${styleString}\n  ${styleKey}: ${this.state.styleValues[styleKey]};`;
    }
    const styleCode = `.${this.previewClass} {${styleString}\n}`;

    return (
      <div className={wrapperStyles.join(' ')}>
        <div className={styles.Pattern}>
          <Pattern code={styleCode}
            title='Styles'
            onCopy={this.openCopyHint}
          />
        </div>
        <div className={styles.Preview}>
          <Preview
            previewClass={this.settings.className}
            template={this.settings.template}
            styles={this.state.styleValues} />
        </div>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    settings: state.editors.settings,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    openHint: (text) => {
      dispatch(actions.openHint(text));
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(OpenedSavingItem);