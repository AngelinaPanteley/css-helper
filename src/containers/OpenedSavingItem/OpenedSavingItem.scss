@import '../../variables';

.Wrapper {
  @include flex-wrapper;

  border-bottom: 2px solid $color-border;
  padding: 10px 30px;
}

.Preview {
  width: 49%;
  display: flex;
  justify-content: center;
}

.Pattern {
  width: 49%;
  margin-bottom: -60px;
}