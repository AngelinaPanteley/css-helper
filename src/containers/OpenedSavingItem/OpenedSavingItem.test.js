"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import { MemoryRouter, Link } from "react-router-dom";
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import OpenedSavingItem from './OpenedSavingItem.js';
import reducer from '../../store/reducers/index';

configure({ adapter: new Adapter() });
const composeEnhancers = process.env.NODE_ENV === 'development' ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ : null || compose;

const store = createStore(reducer, composeEnhancers(
  applyMiddleware(thunk)
));

describe('>>>OpenedSavingItem', () => {

  const props = {
    isOpen: true,
    item: {
      editorName: 'filter',
    }
  }

  const openedSavingItemComponent = <Provider store={store}>
    <MemoryRouter>
      <OpenedSavingItem {...props} />
    </MemoryRouter>
  </Provider>;

  const component = renderer.create(openedSavingItemComponent);

  it('renders itself correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });
});
