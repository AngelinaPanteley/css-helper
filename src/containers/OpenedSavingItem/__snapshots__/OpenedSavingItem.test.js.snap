// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>OpenedSavingItem renders itself correctly 1`] = `
<div
  className=" "
>
  <div>
    <div>
      <p>
        <span>
          Styles
        </span>
        <button
          onClick={[Function]}
        >
          <i
            className="undefined undefined"
          />
        </button>
      </p>
      <pre>
        <code>
          .undefined {
}
        </code>
      </pre>
    </div>
  </div>
  <div>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
  </div>
</div>
`;
